packages/pieces/community/github/src/lib/common/index.ts

```diff
@@ -1,180 +1,120 @@
-						disabled: true,
-						options: [],
-						placeholder: 'please authenticate first and select repo',
-					};
-				}
-				const authProp: OAuth2PropertyValue = auth as OAuth2PropertyValue;
-				const { owner, repo } = repository as RepositoryProp;
-				const labels = await listIssueLabels(authProp, owner, repo);
-				return {
-					disabled: false,
-					options: labels.map((label) => {
-						return {
-							label: label.name,
-							value: label.name,
-						};
-					}),
-				};
-			},
-		}),
+      required,
+      options: async ({ auth, repository }) => {
+        if (!auth || !repository) {
+          return {
+            disabled: true,
+            options: [],
+            placeholder: 'please authenticate first and select repo',
+          };
+        }
+        const authProp: OAuth2PropertyValue = auth as OAuth2PropertyValue;
+        const { owner, repo } = repository as RepositoryProp;
+        const assignees = await getAssignee(authProp, owner, repo);
+        return {
+          disabled: false,
+          options: assignees.map((assignee) => {
+            return {
+              label: assignee.login,
+              value: assignee.login,
+            };
+          }),
+        };
+      },
+    }),
+  labelDropDown: (required = false) =>
+    Property.MultiSelectDropdown({
+      displayName: 'Labels',
+      description: 'Labels for the Issue',
+      refreshers: ['repository'],
+      required,
+      options: async ({ auth, repository }) => {
+        if (!auth || !repository) {
+          return {
+            disabled: true,
+            options: [],
+            placeholder: 'please authenticate first and select repo',
+          };
+        }
+        const authProp: OAuth2PropertyValue = auth as OAuth2PropertyValue;
+        const { owner, repo } = repository as RepositoryProp;
+        const labels = await listIssueLabels(authProp, owner, repo);
+        return {
+          disabled: false,
+          options: labels.map((label) => {
+            return {
+              label: label.name,
+              value: label.name,
+            };
+          }),
+        };
+      },
+    }),
 };
 
-async function getUserRepo(authProp: OAuth2PropertyValue): Promise<GithubRepository[]> {
-	let page = 1;
-	let hasNext;
-	const repos: GithubRepository[] = [];
-	do {
-		const request: HttpRequest = {
-			method: HttpMethod.GET,
-			url: `${githubCommon.baseUrl}/user/repos`,
-			queryParams: {
-				page: page.toString(),
-				per_page: '100',
-			},
-			authentication: {
-				type: AuthenticationType.BEARER_TOKEN,
-				token: authProp.access_token,
-			},
-		};
-
-		const response = await httpClient.sendRequest<GithubRepository[]>(request);
-		repos.push(...response.body);
-
-		hasNext = response.headers?.link?.includes('rel="next"');
-		page += 1;
-	} while (hasNext);
-
-	return repos;
+async function getUserRepo(authProp: OAuth2PropertyValue) {
+  const client = new Octokit({ auth: authProp.access_token });
```

Comment:
@AdamSelene can you use pagination for these dropdown? You can check it [here](https://github.com/octokit/octokit.js?tab=readme-ov-file#pagination)

Reply:
Good catch, sorry I missed it!